server/src/server.rs

```diff
@@ -319,6 +319,9 @@ impl<Q: QueryExecutor + 'static> Builder<Q> {
             .context(MissingSchemaConfigProvider)?;
         let rpc_services = grpc::Builder::new()
             .endpoint(Endpoint::new(self.config.bind_addr, self.config.grpc_port).to_string())
+            .local_endpoint(
+                Endpoint::new(self.config.cluster.node.addr, self.config.grpc_port).to_string(),
```

Comment:
Does this mean we have to config every node ip in order to avoid forward to itself?

This seems not easy to configure.

Reply:
> Does this mean we have to config every node ip in order to avoid forward to itself?

Yes.

> This seems not easy to configure.

Actually, it is easy. Firstly, the forwarding is disabled by default. Secondly, the forwarding makes sense only when run ceresdb in cluster mode, and the `config.cluster.node.addr` must be provided which will be used as the forwarder's local endpoint if we start the ceresdb in cluster mode. In conclusion, the local endpoint of forwarder won't introduce extra configurations besides its own configurations.

Reply:
Correct me if I misunderstand something

The current request flow is:
```
sdk -> route -> server
```
After your patches, the request flow will be this?
```
sdk -> route -> server -> forwarded server
```
I couldn't quite understand why this forward is required.

Reply:
The reason has been explained in https://github.com/CeresDB/ceresdb/issues/147:
> In some scenarios where CeresDB servers are not accessible for client nodes because of the network isolation, the current end-to-end communications won't work any more.

Reply:
Then it sound like this?
```
sdk -> server(served as Querier) -> route -> forwarded server
```

Reply:
Yes. I will update the comment by appending the diagram to it.